pageserver/benches/bench_ingest.rs

```diff
@@ -46,13 +53,24 @@ enum WriteDelta {
     No,
 }
 
+#[derive(serde::Serialize, Clone, Copy, Debug)]
+enum ConcurrentReads {
+    Yes,
+    No,
+}
+
 async fn ingest(
     conf: &'static PageServerConf,
     put_size: usize,
     put_count: usize,
     key_layout: KeyLayout,
     write_delta: WriteDelta,
+    concurrent_reads: ConcurrentReads,
 ) -> anyhow::Result<()> {
+    if matches!(concurrent_reads, ConcurrentReads::Yes) {
+        assert!(matches!(key_layout, KeyLayout::Sequential));
```

Comment:
nit: could just `derive(PartialEq)` and use `==`.

Reply:
https://github.com/neondatabase/neon/pull/12151/commits/5f0dea1e97bdafb42103b260aa62207a2d4cb110